pysatSpaceWeather/instruments/methods/kp_ap.py

```diff
@@ -47,20 +46,344 @@ def references(name, tag):
+                 ap_inst.meta.labels.name: 'Ap',
+                 ap_inst.meta.labels.desc: "daily Ap index",
+                 ap_inst.meta.labels.plot: "Ap",
+                 ap_inst.meta.labels.axis: "Ap",
+                 ap_inst.meta.labels.scale: 'linear',
+                 ap_inst.meta.labels.min_val: 0,
+                 ap_inst.meta.labels.max_val: 400,
+                 ap_inst.meta.labels.fill_val:
+                 ap_inst.meta[ap_name][ap_inst.meta.labels.fill_val],
+                 ap_inst.meta.labels.notes: ''.join(['Ap daily mean calculated',
+                                                     ' from 3-hourly ap ',
+                                                     'indices'])}
+
+    ap_inst.meta[daily_name] = meta_dict
+
+    return
+
+
+def filter_geomag(inst, min_kp=0, max_kp=9, filter_time=24, kp_inst=None):
+    """Filters pysat.Instrument data for given time after Kp drops below gate.
+
+    Parameters
+    ----------
+    inst : pysat.Instrument
+        Instrument with non-Kp data to be filtered by geomagnetic activity
+    min_kp : float
+        Minimum Kp value allowed. Kp values below this filter the data in
+        inst (default=0)
+    max_kp : float
+        Maximum Kp value allowed. Kp values above this filter the data in
+        inst (default=9)
+    filter_time : int
+        Number of hours to filter data after Kp drops below max_kp (default=24)
+    kp_inst : pysat.Instrument or NoneType
+        Kp pysat.Instrument object with or without data already loaded. If None,
+        will load GFZ historic kp data for the instrument date (default=None)
+
+    Note
+    ----
+    Loads Kp data for the same timeframe covered by inst and sets inst.data to
+    NaN for times when Kp > max_kp or Kp < min_kp and for filter_time after Kp
+    drops below max_kp.
+
+    Default max and min values accept all Kp, so changing only one will cause
+    the filter to act as a high- or low-pass function.
+
+    This routine is written for standard Kp data (tag=''), not the forecast or
+    recent data.  However, it will work with these Kp data if they are supplied.
+
+    This routine is designed to be used with the 'modify' flag if applied as
+    a custom routine.
+
+    """
+    # Load the desired data
+    if kp_inst is None:
+        kp_inst = pysat.Instrument(inst_module=pysat_sw.instruments.sw_kp,
+                                   tag='', pad=pds.DateOffset(days=1))
+
+    if kp_inst.empty:
+        kp_inst.load(date=inst.date, verifyPad=True)
+
+    # Begin filtering, starting at the beginning of the instrument data
+    sel_data = kp_inst[(inst.date - pds.DateOffset(days=1)):
+                       (inst.date + pds.DateOffset(days=1))]
+    ind, = np.where((sel_data['Kp'] > max_kp) | (sel_data['Kp'] < min_kp))
+    for lind in ind:
+        sind = sel_data.index[lind]
+        eind = sind + pds.DateOffset(hours=filter_time)
+        inst.data[sind:eind] = np.nan
```

Comment:
Going through the Instrument itself provides (in theory!) the most data compatible interface.

```suggestion
        inst[sind:eind] = np.nan
```